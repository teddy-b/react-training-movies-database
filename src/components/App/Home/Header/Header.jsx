import React from 'react'
import PropTypes from 'prop-types'

import SearchBar from './SearchBar'
import SortInfo from './SortInfo'

const Header = ({
  count,
  onSearch,
  onSearchMoviesByGenre,
  onSearchMoviesByTitle,
  onSortMoviesByRating,
  onSortMoviesByRelaseDate,
  searchBy,
  sortBy
}) => (
  <header className="header">
    <SearchBar
      onSearch={onSearch}
      onSearchMoviesByGenre={onSearchMoviesByGenre}
      onSearchMoviesByTitle={onSearchMoviesByTitle}
      searchBy={searchBy}
    />
    <SortInfo
      count={count}
      onSortMoviesByRating={onSortMoviesByRating}
      onSortMoviesByRelaseDate={onSortMoviesByRelaseDate}
      sortBy={sortBy}
    />
  </header>
)

Header.propTypes = {
  count: PropTypes.number.isRequired,
  onSearch: PropTypes.func.isRequired,
  onSearchMoviesByGenre: PropTypes.func.isRequired,
  onSearchMoviesByTitle: PropTypes.func.isRequired,
  onSortMoviesByRating: PropTypes.func.isRequired,
  onSortMoviesByRelaseDate: PropTypes.func.isRequired,
  searchBy: PropTypes.string.isRequired,
  sortBy: PropTypes.string.isRequired
}

export default Header
