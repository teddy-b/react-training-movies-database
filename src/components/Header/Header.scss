.searchBar {
  background-color: dimgrey;
  color: white;
  padding: .5em;
}

.searchBarTitle {
  text-transform: uppercase;
}

.searchInput {
  width: 98%;
  padding: .8em;
  background-color: black;
  color: inherit;
  border: none;
  border-bottom: 1px solid red;
}
.searchFooter {
  padding: .5em;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.searchBy {
  text-transform: uppercase;
}

.searchByBtns {
  flex-grow: 2;
}

.searchByTitleBtn, .searchByGenreBtn, .searchBtn {
  margin: .3em;
  padding: .6em;
  background-color: red;
  color: white;
  border: none;
}

.searchBtn {
  margin: .2em;
  padding: .8em;
  width: 15vw;
  background-color: red;
  color: white;
  border: none;
}

.sorting {
  display: flex;
  justify-content: space-between;
  background-color: lightgrey;
  padding: .5em;

  a {
    margin: .5em;
  }
}
