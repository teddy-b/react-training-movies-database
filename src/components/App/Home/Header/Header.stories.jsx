/* @flow */

import * as React from 'react'
import { MemoryRouter } from 'react-router-dom'

import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'

import type { Renderable } from '@storybook/react'

import Header from './Header'
import { SEARCH_BY, SORT_BY } from '../../../../constants/global'

const props = {
  count: 24,
  onSearch: action('clicked'),
  onSearchMoviesByGenre: action('clicked'),
  onSearchMoviesByTitle: action('clicked'),
  onSortMoviesByRating: action('clicked'),
  onSortMoviesByRelaseDate: action('clicked'),
  searchBy: SEARCH_BY.title,
  sortBy: SORT_BY.releaseDate
}

storiesOf('Components/Header', module)
  .add('default', (): Renderable => (
    <MemoryRouter>
      <Header {...props} />
    </MemoryRouter>
  ))
