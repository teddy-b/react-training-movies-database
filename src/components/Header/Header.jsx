import React from 'react'

import Logo from '../Logo'
import './Header.scss'

const Header = ({ count, onSearch, onSort }) => (
  <header className="header">
    <div className="searchBar">
      <Logo />
      <p className="searchBarTitle">Find your movie</p>
      <input className="searchInput" type="text" />
      <div className="searchFooter">
        <div className="searchBy">Search by </div>
        <div className="searchByBtns">
          <button className="searchByTitleBtn" onClick={() => onSearch('title')}>Title</button>
          <button className="searchByGenreBtn" onClick={() => onSearch('genre')}>Genre</button>
        </div>
        <button className="searchBtn" onClick={() => onSearch('title')}>Search</button>
      </div>
    </div>
    <div className="sorting">
      {count > 0 && <div>{count} movies found</div>}
      {count > 0 && <div>
        <span>Sort by </span>
        <a onClick={() => onSort('release date')}>Release date</a> 
        <a onClick={() => onSort('rating')}>Rating</a>
      </div>}
    </div>
  </header>
)

export default Header
