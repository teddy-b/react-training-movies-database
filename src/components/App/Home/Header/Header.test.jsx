import React from 'react'
import renderer from 'react-test-renderer'

import Header from './Header'

jest.mock('./SearchBar', () => 'SearchBar')
jest.mock('./SortInfo', () => 'SortInfo')

describe('Header', () => {
  const props = {
    count: 10,
    onSearch: jest.fn(),
    onSearchMoviesByGenre: jest.fn(),
    onSearchMoviesByTitle: jest.fn(),
    onSortMoviesByRating: jest.fn(),
    onSortMoviesByRelaseDate: jest.fn(),
    searchBy: 'title',
    sortBy: 'release_date'
  }

  it('renders correctly', () => {
    const component = renderer.create(<Header {...props} />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
