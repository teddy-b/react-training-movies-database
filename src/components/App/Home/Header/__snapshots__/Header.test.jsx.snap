// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header>
  <SearchBar
    onSearch={[MockFunction]}
    onSearchMoviesByGenre={[MockFunction]}
    onSearchMoviesByTitle={[MockFunction]}
    searchBy="title"
  />
  <SortInfo
    count={10}
    onSortMoviesByRating={[MockFunction]}
    onSortMoviesByRelaseDate={[MockFunction]}
    sortBy="release_date"
  />
</header>
`;
